CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc

```diff
@@ -238,126 +238,123 @@ void HcaluLUTTPGCoder::update(const HcalDbService& conditions) {
-	    CaloSamples samples(cell, 1);
-
-	    int granularity = meta->getLutGranularity();
-
-	    for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
-	      frame.setSample(0,HcalQIESample(adc));
-	      coder.adc2fC(frame,samples);
-	      float adc2fC = samples[0];
-
-	      if (isMasked) inputLUT_[lutId][adc] = 0;
-	      else inputLUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE8_LUT_BITMASK);
-	    }
-
-            unsigned short data[] = {0, 0, 0};
-            QIE11DataFrame upgradeFrame(edm::DataFrame(0, data, 3));
-            CaloSamples upgradeSamples(cell, 1);
-            for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
-               upgradeFrame.setSample(0, adc, 0, true);
-               coder.adc2fC(upgradeFrame, upgradeSamples);
-               float adc2fC = upgradeSamples[0];
-
-               if (isMasked) {
-                  upgradeQIE11LUT_[lutId][adc] = 0;
-               } else {
-                  upgradeQIE11LUT_[lutId][adc] = (LutElement) std::min(std::max(0, int((adc2fC -ped) * gain * rcalib / nominalgain_ / granularity)), QIE11_LUT_BITMASK);
-                  if (adc >= mipMin and adc < mipMax)
-                     upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB0;
-                  else if (adc >= mipMax)
-                     upgradeQIE11LUT_[lutId][adc] |= QIE11_LUT_MSB1;
-               }
-            }
-	  }  // endif HBHE
-	  else if (subdet == HcalForward){
-             HFDataFrame frame(cell);
-             frame.setSize(1);
-             CaloSamples samples(cell, 1);
-
-             for (unsigned int adc = 0; adc < INPUT_LUT_SIZE; ++adc) {
-                frame.setSample(0,HcalQIESample(adc));
-                coder.adc2fC(frame,samples);
-                float adc2fC = samples[0];
-                if (isMasked) inputLUT_[lutId][adc] = 0;
-                else inputLUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[abs(ieta)] )), QIE8_LUT_BITMASK);
-             }
-
-             unsigned short data[] = {0, 0, 0, 0};
-             QIE10DataFrame upgradeFrame(edm::DataFrame(0, data, 4));
-             CaloSamples upgradeSamples(cell, 1);
-             for (unsigned int adc = 0; adc < UPGRADE_LUT_SIZE; ++adc) {
-                upgradeFrame.setSample(0, adc, 0, 0, 0, true);
-                coder.adc2fC(upgradeFrame, upgradeSamples);
-                float adc2fC = upgradeSamples[0];
-
-                if (isMasked)
-                   upgradeQIE10LUT_[lutId][adc] = 0;
-                else
-                   upgradeQIE10LUT_[lutId][adc] = std::min(std::max(0,int((adc2fC - ped) * gain * rcalib / lsb_ / cosh_ieta[abs(ieta)] )), QIE10_LUT_BITMASK);
-             }
-	  } // endif HF
-
-	} // for depth
-      } // for iphi
-    } // for iphi
-  }// for subdet
+  for (const auto& id: metadata->getAllChannels()) {
+     if (not (id.det() == DetId::Hcal and topo_->valid(id)))
```

Comment:
ok, thanks - is a dangerous condition to leave around, so lets try to clean up after than plan1 issues are behind us